### ISSUE TITLE
[@angular-eslint/template/i18n] reports false-positives and false-negatives

### ISSUE THREAD

**Issue by EmmanuelRoux** [angular-eslint/angular-eslint#710]
**Description and reproduction of the issue**

Rule `@angular-eslint/template/i18n` is unexpectedly reporting errors for some texts, and does not report errors for texts it would be expected to.

```JSON
{
  "rules": {
    "@angular-eslint/template/i18n": [
      "error",
      {
        "checkId": false
      }
    ]
  }
}
```

```HTML
<div>this is reported (as expected)</div>
<div i18n>this is not reported (as expected)</div>
<div>this: {{shouldReportError}}</div> <!-- this should be reported, but is not -->
<div>{{shouldNotReportError}}</div> <!-- this should not be reported, but it is -->
```

Running `npx eslint "src/app/**"` produces following output for `src/app/app.component.ts`:

![image](https://user-images.githubusercontent.com/1926413/136243246-c6bb2836-ab76-4257-8038-fee10bae346c.png)

Full code to reproduce the problem can be found on Stackblitz: https://stackblitz.com/edit/angular-ivy-nteny3

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `12.5.0` |
| `@angular-eslint/template-parser`         | `12.5.0` |
| `@typescript-eslint/parser`               | `4.28.2` |
| `ESLint`                                  | `7.26.0` |
| `node`                                    | `14.17.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 12.2.8
Node: 14.17.0
Package Manager: npm 6.14.13
OS: win32 ia32

Angular: 12.2.8
... animations, cli, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1202.8
@angular-devkit/build-angular   12.2.8
@angular-devkit/core            12.2.8
@angular-devkit/schematics      12.2.8
@schematics/angular             12.2.8
rxjs                            6.6.7
typescript                      4.2.4
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by EmmanuelRoux** [comment]
Hi @rafaelss95, I've already included a link to Stackblitz in the description (https://stackblitz.com/edit/angular-ivy-nteny3)
